orangecontrib/text/widgets/utils/readers.py

```diff
@@ -0,0 +1,215 @@
+import fnmatch
+import os
+import pathlib
+import re
+
+from collections import namedtuple
+from string import capwords
+from types import SimpleNamespace as namespace
+
+import docx2txt
+from odf.opendocument import load
+from odf import text, teletype
+
+from pdfminer.pdfparser import PDFParser, PDFDocument
+from pdfminer.pdfinterp import PDFResourceManager, PDFPageInterpreter
+from pdfminer.converter import PDFPageAggregator
+from pdfminer.layout import LAParams, LTTextBox, LTTextLine
+from bs4 import BeautifulSoup
+
+
+TextData = namedtuple(
+    "Text",
+    ["name", "path", "ext", "category", "content"]
+)
+TextData.isvalid = property(lambda self: True)
+
+TextDataError = namedtuple(
+    "TextDataError",
+    ["path", "error", "error_str"]
+)
+TextDataError.isvalid = property(lambda self: False)
+
+
+class Reader():
+    def __init__(self, path):
+        self.path = path
+        self.replace_white_space = False
+        self.content = None
+
+    @classmethod
+    def get_reader(cls, path):
+        ext = pathlib.Path(path).suffix
+        return eval("{}Reader(path)".format(capwords(ext[1:])))
+
+    def read(self, replace_white_space=False):
+        try:
+            textdata = self.read_file()
+        except:
```

Comment:
The same happens for [webtext corpus](https://github.com/nltk/nltk_data/blob/gh-pages/packages/corpora/webtext.zip). Out of 6 text files, I only got two if TxtReader is in text mode.